synapse/app/homeserver.py

```diff
@@ -259,6 +273,13 @@ def quit_with_error(error_string):
     sys.exit(1)
 
 
+def check_bind_error(e, address, bind_addresses):
+    if address == '0.0.0.0' and '::' in bind_addresses:
+        logger.warn('Failed to listen on 0.0.0.0, continuing because listening on [::]')
```

Comment:
could you add a comment about why this is a sensible thing to do (ie, explaining some of the problems discussed in #2232 ) ?